src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs

```diff
@@ -0,0 +1,279 @@
+            // to execute on the client.
+            // 2) There is also a bug (same tracking item) where code actions that edit documents other than the
+            // one where the code action was invoked from do not work. We must temporarily execute these as commands
+            // as well.
+            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+            var runAsCommand = false;
+
+            var applyChangesOperations = operations.Where(operation => operation is ApplyChangesOperation);
+            if (applyChangesOperations.Any())
+            {
+                var workspaceEdits = await ComputeWorkspaceEdits(applyChangesOperations, document!, cancellationToken).ConfigureAwait(false);
+                if (workspaceEdits.Any())
+                {
+                    codeAction.Edit = new LSP.WorkspaceEdit { DocumentChanges = workspaceEdits };
+                }
+                else
+                {
+                    // The workspace edit is something we don't currently support, like adding a new document.
+                    runAsCommand = true;
+                }
+            }
+
+            // Set up to run as command on the server instead of using WorkspaceEdits.
+            var commandOperations = operations.All(operation => !(operation is ApplyChangesOperation));
+            if (commandOperations || runAsCommand)
+            {
+                codeAction.Command = SetCommand(codeAction, data);
+            }
+
+            return codeAction;
+
+            // Local functions
+            static async Task<TextDocumentEdit[]> ComputeWorkspaceEdits(
+                IEnumerable<CodeActionOperation> applyChangesOperations,
+                Document document,
+                CancellationToken cancellationToken)
+            {
+                using var _ = ArrayBuilder<TextDocumentEdit>.GetInstance(out var textDocumentEdits);
+                foreach (ApplyChangesOperation applyChangesOperation in applyChangesOperations)
+                {
+                    var solution = document.Project.Solution;
+                    var changes = applyChangesOperation.ChangedSolution.GetChanges(solution);
+                    var projectChanges = changes.GetProjectChanges();
+
+                    // TO-DO: If the change involves adding a document, execute via command instead of WorkspaceEdit
+                    // until adding documents is supported in LSP: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // After support is added, remove the below if-statement and add code to support adding documents.
+                    var addedDocuments = projectChanges.SelectMany(
+                        pc => pc.GetAddedDocuments().Concat(pc.GetAddedAdditionalDocuments().Concat(pc.GetAddedAnalyzerConfigDocuments())));
+                    if (addedDocuments.Any())
+                    {
+                        return textDocumentEdits.ToArray();
```

Comment:
It seems like if there are already textDocumentEdits in the array, then the check on line 105 will execute and will never hit the command.  Should this return an empty array?